README.md.template

```diff
@@ -0,0 +1,23 @@
+# ShellCheck pre-commit hook
+
+This is NOT an official [pre-commit hook](https://pre-commit.com/) for
+[differential-shellcheck](https://github.com/redhat-plumbers-in-action/differential-shellcheck,
+the static analysis tool for changed shell scripts.
+
+Activate by adding it to your `.pre-commit-config.yaml`:
+
+```sh
+repos:
+-   repo: https://github.com/mpoberezhniy/differential-shellcheck-precommit
+    rev: $VERSION
+    hooks:
+    -   id: differential-shellcheck
+#       args: ["--severity=warning"]  # Optionally only show errors and warnings
+```
+
+#### Why a separate repo?
+
+This repo keeps the pre-commit hook out of the critical path of differential-shellcheck
+releases, reducing the number of things that can go wrong. This in turn helps
+ensure a smoother `pre-commit autoupdate`. 
```

Comment:
I'm not a pre-commit user, but I think you are specifying the exact version you are using, so unless you update to a potentially broken version, everything should work.

I'm not against having a separate repo. On the other hand, having it in one repo would make it easier to add a set of tests that could prevent the pre-commit hook from breaking. But there would have to be certainly some level of cooperation either way :-)

Reply:
As of my `pre-commit autoupdate` code understanding, it pulls a repo copy locally (a precommit repo to be specific), iterates over tags, and selects the newest version to use. I think if this repo has proper actions to build release images - it should be stable.

I am open to merging this repository to differential-shellcheck or moving it somewhere else.
Thinking of the single responsibility principle we can keep them separate, I think I will have to implement an analog of `src/summary.sh` to get the output more readable in a terminal. This should allow not implementing redundant flags to control output styling and not adding the logic to stash/pop changes instead of checking out base/head commits.

Reply:
I agree that some work needs to be done to make differential shellcheck accessible from outside of GitHub Actions. My ultimate goal would be to make it possible to package it into distributions for local work.

- https://github.com/redhat-plumbers-in-action/differential-shellcheck/issues/179

I want to thank you for working on pre-commit; it's awesome. I will leave you the choice of where to have it (pre-commit code). :-)

Reply:
I think I will be able to refactor it to have pre-commit and post-receive modes so `pre-commit` will be able to use it and the action will use it in a post-receive mode. This should help to get rid of the whole pre-commit.sh file and handling everything in index.sh while keeping the code readable.

If we have it as hooks - this should allow other SCM users to define their server-side hooks to trigger CI/CD and lint differentially.